java/service_instance.go

```diff
@@ -1577,3 +1578,72 @@ func (c *ServiceInstanceClient) waitForServiceInstanceDeleted(input *GetServiceI
+type ScaleUpDownServiceInstanceInput struct {
+	// Groups properties for the Oracle WebLogic Server component (WLS).
+	// Required
+	Components ScaleUpDownComponent `json:"components"`
+	// Name of the Java Cloud Service instance.
+	// Required.
+	Name string `json:"-"`
+}
+
+// ScaleUpDownComponent defines the attributes for the WebLogic Server components when scaling up and down
+// the service instance
+type ScaleUpDownComponent struct {
+	// Properties for the Oracle WebLogic Server (WLS) component.
+	// Required
+	WLS ScaleUpDownWLS `json:"WLS"`
+}
+
+// ScaleUpDownWLS defines the properties for the Oracle WebLogic Server (WLS) component.
+type ScaleUpDownWLS struct {
+	// A single host name. Only application cluster hosts can be specified.
+	// Required
+	Hosts []string `json:"hosts"`
+	// Flag that indicates whether to ignore Managed Server heap validation (true) or perform heap
+	// validation (false) before a scale down request is accepted. Default is false.
+	// When the flag is not set or is false, heap validation is performed before scaling.
+	// If a validation error is not generated, the Managed Server JVM is restarted with the new shape
+	// after scaling down.
+	// When the flag is true, heap validation is not performed. Before you set the flag to true,
+	// make sure the -Xms value is low enough for the Managed Server JVM to restart on the new shape
+	// after scaling down. The -Xms value should be lower than one-fourth the size of the memory
+	// associated with the shape. Use the WebLogic Server Administration Console to edit the value in
+	// the server start arguments, if necessary.
+	// Optional
+	IgnoreManagedServerHeapError bool `json:"ignoreManagedServerHeapError,omitempty"`
+	// Desired compute shape for the target host.
+	// Required
+	Shape ServiceInstanceShape `json:"shape"`
+}
+
+// ScaleUpDownServiceInstance scales the service instance up or down depending on the shape passed in
+func (c *ServiceInstanceClient) ScaleUpDownServiceInstance(input *ScaleUpDownServiceInstanceInput) error {
+	if c.PollInterval == 0 {
+		c.PollInterval = waitForServiceInstanceReadyPollInterval
+	}
+	if c.Timeout == 0 {
+		c.Timeout = waitForServiceInstanceReadyTimeout
+	}
+
+	if err := c.updateResource(input.Name, serviceInstanceScaleUpDownpath, "POST", input); err != nil {
+		return fmt.Errorf("unable to update Java Service Instance %q: %+v", input.Name, err)
+	}
+
+	// Call wait for instance ready now, as updating the instance is an eventually consistent operation
+	getInput := &GetServiceInstanceInput{
+		Name: input.Name,
+	}
+
+	// Wait for the service instance to be running and return the result
+	// Don't have to unqualify any objects, as the GetServiceInstance method will handle that
+	serviceInstance, err := c.WaitForServiceInstanceRunning(getInput, c.PollInterval, c.Timeout)
+	// If the service instance is returned as nil if it enters a terminating state.
```

Comment:
I think this should be "The service instance is returned as nil if it enters a terminating state."